### PR TITLE
Support for bazel buildsystem

### PR DESCRIPTION
**Pull request by mjcarroll**

### CODE REVIEW

**Review by JShep1 — COMMENTED** [review]
Initial pass, but can't test with cmake yet due to the `ign_plugin` having a failing build.  All looks good on the bazel side, I'm having one failing test: `RenderingIFace_TEST.cc`.  Looks like it's due to an undefined symbol in `libignition-rendering-ogre2.so`: `FT_Init_FreeType`

**Comment by JShep1** [comment]
> Initial pass, but can't test with cmake yet due to the `ign_plugin` having a failing build. All looks good on the bazel side, I'm having one failing test: `RenderingIFace_TEST.cc`. Looks like it's due to an undefined symbol in `libignition-rendering-ogre2.so`: `FT_Init_FreeType`

For the record, I replicated this issue on my desktop running Bionic

**Review by mahiuchun — CHANGES_REQUESTED** [review]
The latest commit fails to build for me.

```
ign_rendering/ogre2/src/Ogre2DepthCamera.cc:41:10: fatal error: Ogre2ParticleNoiseListener.hh: No such file or directory
 #include "Ogre2ParticleNoiseListener.hh"
```

File: examples/ogre2_demo/Main.cc [hunk 1]
```diff
@@ -31,16 +31,14 @@
...
 #include "GlutWindow.hh"
 
 using namespace ignition;
 using namespace rendering;
 
 
 const std::string RESOURCE_PATH =
-    common::joinPaths(std::string(PROJECT_BINARY_PATH), "media");
-
+    common::joinPaths(std::string("ign_rendering/examples/ogre2_demo"), "media");
```
Comment (chapulina) [inline examples/ogre2_demo/Main.cc:41]: This assumes the user is running the example from some specific directory

File: include/ignition/rendering/RenderPassSystem.hh [hunk 2]
```diff
@@ -89,24 +89,21 @@ namespace ignition
...
-    static classname##Factory global_##classname##Factory;
     }
   }
 }
+
+
+/// \brief Render pass registration macro
+///
+/// Use this macro to register render pass with the render pass factory.
+#define IGN_RENDERING_REGISTER_RENDER_PASS(classname, interface) \
```
Comment (chapulina) [inline include/ignition/rendering/RenderPassSystem.hh:100]: This move doesn't seem backwards-compatible

**Review by chapulina — CHANGES_REQUESTED** [review]
CI is unhappy

**Comment by chapulina** [comment]
I believe this PR is outdated and not being worked on. Please reopen if this starts being worked on again.

**Pull request closed without merging**